qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,383 @@
-                'last_artifact': 'TODO new gui'
+
+    if investigation_type:
+        investigation_type = _process_investigation_type(
+            investigation_type, user_defined_investigation_type,
+            new_investigation_type)
+        prep.investigation_type = investigation_type
+
+    msg = ''
+    status = 'success'
+    if prep_template:
+        fp = check_fp(study_id, prep_template)
+        if fp['status'] != 'success':
+            # Unknown filepath, so return the error message
+            return fp
+        fp = fp['file']
+        try:
+            with warnings.catch_warnings(record=True) as warns:
+                pt = PrepTemplate(int(prep_id))
+                df = load_template_to_dataframe(fp)
+                pt.extend(df)
+                pt.update(df)
+                remove(fp)
+
+                # join all the warning messages into one. Note that this info
+                # will be ignored if an exception is raised
+                if warns:
+                    msg = '\n'.join(set(str(w.message) for w in warns))
+                    status = 'warning'
+
+        except Exception as e:
+            # Some error occurred while processing the sample template
+            # Show the error to the user so they can fix the template
+            status = 'error'
+            msg = str(e)
+    return {'status': status,
+            'message': msg,
+            'file': prep_template}
+
+
+def prep_template_delete_req(prep_id, user_id):
+    """Delete the prep template
+
+    Parameters
+    ----------
+    prep_id : int
+        The prep template to update
+    user_id : str
+        The current user object id
+
+    Returns
+    -------
+    dict of str
+        {'status': status,
+         'message': message}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+
+    msg = ''
+    status = 'success'
+    try:
+        PrepTemplate.delete(prep.id)
+    except Exception as e:
+        msg = ("Couldn't remove prep template: %s" % str(e))
+        status = 'error'
+
+    return {'status': status,
+            'message': msg}
+
+
+@execute_as_transaction
+def prep_template_filepaths_get_req(prep_id, user_id):
+    """Returns all filepaths attached to a prep template
+
+    Parameters
+    ----------
+    prep_id : int
+        The current prep template id
+    user_id : int
+        The current user object id
+
+    Returns
+    -------
+    dict of objects
+        {'status': status,
+         'message': message,
+         'filepaths': [(filepath_id, filepath), ...]}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    return {'status': 'success',
+            'message': '',
+            'filepaths': prep.get_filepaths()
+            }
+
+
+def prep_ontology_get_req():
```

Comment:
Because this is the ENA ontology that is already attached to the prep template and prep template page currently.